### ISSUE TITLE
Zig and Rust article malloc and stack mismatch

### ISSUE THREAD

**Issue by grumbach** [matklad/matklad.github.io#169]
```
no programs understand what their stack usage is exactly
...
When we call malloc, we just hope that we have enough stack space for it, we almost never check.
```

Malloc is heap allocation not stack allocation. You probably meant `memory space` and not `stack space`.

**Comment by matklad** [comment]
No, this is correct. Like any function, Malloy needs _some_ amount of stack space for its local variables. Moreover, the stack usage of malloc tends to be comparatively high due to things like 

https://github.com/aerospike/jemalloc/blob/05108b5010a511226fb7586543f4162dd2d31d2b/include/jemalloc/internal/rb.h#L467

**Comment by grumbach** [comment]
Oh ok, I indeed didn't think of it that way 👍
